tools/internal_ci/helper_scripts/check_python_artifacts_size.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+# Copyright 2024 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Checks if any of the Python artifacts exceeds a certern size limit since
+# Pypi has a per-file size limit.
+
+set -ex
+
+find . -path "*/artifacts/*" -size +80M | egrep '.*' && echo "Found Python artifacts larger than 80 MB." && FAILED="true"
```

Comment:
Since the artifacts are currently over 80MB, we have two options:

1. Temporarily bump this to 110MB and have a TODO (which we actually follow up on) or
2. Wait to merge this PR until we actually fix the file size.

It's up to you which one to go with.

Reply:
Since this PR won't be useful until next release, I'll merge it after we fix the size issue.